sdk/go/common/resource/config/value.go

```diff
@@ -50,77 +44,47 @@ func NewObjectValue(v string) Value {
 // Value fetches the value of this configuration entry, using decrypter to decrypt if necessary.  If the value
 // is a secret and decrypter is nil, or if decryption fails for any reason, a non-nil error is returned.
 func (c Value) Value(decrypter Decrypter) (string, error) {
-	if !c.secure {
+	if decrypter == NopDecrypter {
```

Comment:
Do we need this, NopDecrypter will just return it below won't it?

Reply:
We need this for compat.

The prior implementation took this shortcut for `NopDecrypter` and as a result does not process nested secure strings. For example, given this object value:

```
{"foo": {"secure": "bar"}}
```

`Value(NopDecrypter)` will return the literal text, while `Value(NewBlindingDecrypter())` will replace the entire nested secure string with `[secret]`:

```
{"foo": "[secret]"}
```

Reply:
hmm ok, I've always disliked this specialization on the decrypter type but can think about this more later.